cmdliner-1.2.0/B0.ml

```diff
@@ -0,0 +1,64 @@
+open B0_kit.V000
```

Comment:
`cmdliner` should not be added to our source tree. If you added it to meet the `cmdliner` dependency of in `bin/dune` I suspect you may have skipped a step in setting up your development environment? I suggest

- removing the source of `cmdliner`
- deleting the `_opam` file in the root of this repo (if you have a copy of that)
- setting up your dev environment again following https://github.com/NathanReb/ocaml-api-watch/blob/main/CONTRIBUTING.md#contributing

Reply:
> `cmdliner` should not be added to our source tree. If you added it to meet the `cmdliner` dependency in `bin/dune` I suspect you may have skipped a step in setting up your development environment? I suggest
> 
> * removing the source of `cmdliner`
> * deleting the `_opam` file in the root of this repo (if you have a copy of that)
> * setting up your dev environment again following https://github.com/NathanReb/ocaml-api-watch/blob/main/CONTRIBUTING.md#contributing

I had followed the dev setup environment but faced a lot of issues on my windows. Thus, I believe there occurred some error with cmdliner. I'll look into this.

Reply:
> > `cmdliner` should not be added to our source tree. If you added it to meet the `cmdliner` dependency in `bin/dune` I suspect you may have skipped a step in setting up your development environment? I suggest
> > 
> > * removing the source of `cmdliner`
> > * deleting the `_opam` file in the root of this repo (if you have a copy of that)
> > * setting up your dev environment again following https://github.com/NathanReb/ocaml-api-watch/blob/main/CONTRIBUTING.md#contributing
> 
> I had followed the dev setup environment but faced a lot of issues on my windows. Thus, I believe there occurred some error with cmdliner. I'll look into this. Thank you.

While following the instructions for setting the environment , when I run the command 
'''opam switch create ./ --deps-only -t'''
it gives the following error :

''''opam: unknown option `--deps-only', did you mean `-d' ?
Usage: opam switch [OPTION]... [COMMAND] [ARG]...'''

how should I proceed further ?

Reply:
> `cmdliner` should not be added to our source tree. If you added it to meet the `cmdliner` dependency in `bin/dune` I suspect you may have skipped a step in setting up your development environment? I suggest
> 
> * removing the source of `cmdliner`
> * deleting the `_opam` file in the root of this repo (if you have a copy of that)
> * setting up your dev environment again following https://github.com/NathanReb/ocaml-api-watch/blob/main/CONTRIBUTING.md#contributing

I tried repeating the steps again but when I ran the code 
```
opam switch create ./ --deps-only -t
```
it gave the following error : 
```
opam: unknown option `--deps-only', did you mean `-d' ?
Usage: opam switch [OPTION]... [COMMAND] [ARG]...
Try `opam switch --help' or `opam --help' for more information.
```

Where is it incorrect ?

Reply:
Ah! Good find. You may have an old opam version? Can you show me what you get for your opam version?

```
$ opam --version
2.1.5
```

Reply:
> Ah! Good find. You may have an old opam version? Can you show me what you get for your opam version?
> 
> ```
> $ opam --version
> 2.1.5
> ```

When i run it on powershell , I get the following : 
```
opam --version
2.2.0~alpha0~20221228
```

But when I run it on bash , I get the following : 
```
 opam --version
1.2.2
```

Why is it giving two different versions ?

Reply:
Bash and powershell are two different shell environments. As far as I know, if you are running bash in WSL, it may have a totally different environment of packages from what is available on Windows' powershell.

To get your WSL version op to date, check out https://ocaml.org/docs/ocaml-on-windows#wsl2, which then links to instructions for installing on Linux. You may want to ensure that your Linux distro's package manager repo has been updated (e.g. via `apt updat`) before trying to upgrade or reinstall opam.

Let me know how these efforts go!

Reply:
For running ```apt update``` bash needed administrative permissions, thus I ran the following code ```sudo apt update```. It worked with running ```opam switch create ./ --deps-only -t```.
Thank you for the information and assistance !

Reply:
Awesome! Glad to hear it :)